soc/xtensa/intel_adsp/cavs/power.c

```diff
@@ -78,12 +78,21 @@ __imr void power_init(void)
 	/* Disable power gating for first cores */
 	CAVS_SHIM.pwrctl |= CAVS_PWRCTL_TCPDSPPG(0);
 
+	/* Claim ownership of LPGDMA controller and channel registers */
+	sys_write32(LPGPDMAC_CTLOSEL_FLAG | LPGPDMAC_CHOSEL_FLAG,
+		    ADSP_LPGPDMACxO_ADDRESS(0));
+	sys_write32(LPGPDMAC_CTLOSEL_FLAG | LPGPDMAC_CHOSEL_FLAG,
+		    ADSP_LPGPDMACxO_ADDRESS(1));
+
```

Comment:
dma_intel_adsp_gpdma.c has gpdma_select_owner() doing this, but when ACE was added it was excluded from this code path.

https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/dma/dma_intel_adsp_gpdma.c#L204

I guess this is needed for all gpdma instances regardless of family?

Reply:
Wow, how did I miss that? I did do greps on the registers across whole Zephyr code-tree, but somehow I missed this.

Now I'm not sure what is the correct place. This is really SoC specific programming and relates to how the LPGPDMAC IP is integrated on this specific SoC. Not sure whether the DMA Zephyr driver should be doing these. Plus we have similar programming already for e.g. I2S and DMIC for cAVS2.5.

But yeah, I'm ok both ways just as long somebody programs these. Without this, the peripherals cannot be used (and I want to remove this programming from SOF code for cAVS hw).

Reply:
In cavs this is a little different than ace, this register is not a part of gpdma but it still should be programmed if requested by gpdma so you can leave it as is in dma_intel_adsp_gpdma.c

And it looks like if we have duplicated defines for this register, it would be greate to remove DSP_INIT_LPGPDMA from gpdma driver and use already defined ADSP_LPGPDMACxO_ADDRESS

Reply:
Isn't it already being doing ?
```

		 * expected to succeed in a tight loop to look
		 * for concurrency problems.
		 */
		ret = string_nlen(user_string, BUF_SIZE, &err);
		zassert_equal(err, 0, "user string faulted");
		zassert_equal(ret, strlen(user_string),
			      "incorrect length returned");

		ret = string_alloc_copy("this is a kernel string");
		zassert_equal(ret, 0, "string should have matched");

		ret = string_copy("this is a kernel string");
		zassert_equal(ret, 0, "string should have matched");
```